x-pack/plugins/enterprise_search/server/utils/create_pipeline_definitions.ts

```diff
@@ -225,3 +230,70 @@ export const createIndexPipelineDefinitions = (
   });
   return { created: [indexName, `${indexName}@custom`, `${indexName}@ml-inference`] };
 };
+
+/**
+ * Format the body of an ML inference pipeline for a specified model.
+ * Does not create the pipeline, only returns JSON for the user to preview.
+ * @param modelId modelId selected by user.
+ * @param sourceField The document field that model will read.
+ * @param destinationField The document field that the model will write to.
+ * @param esClient the Elasticsearch Client to use when retrieving model details.
+ */
+export const formatMlPipelineBody = async (
+  modelId: string,
+  sourceField: string,
+  destinationField: string,
+  esClient: ElasticsearchClient
+): Promise<ESPipeline> => {
+  const models = await esClient.ml.getTrainedModels({ model_id: modelId });
+  // if we didn't find this model, we can't return anything useful
+  if (models.trained_model_configs === undefined || models.trained_model_configs.length === 0) {
+    throw new Error(`Couldn't find any trained models with id [${modelId}]`);
+  }
+  const model = models.trained_model_configs[0];
+  // if model returned no input field, insert a placeholder
+  let modelInputField = 'MODEL_INPUT_FIELD';
+  if (
+    model.input !== undefined &&
+    model.input.field_names !== undefined &&
+    model.input.field_names.length > 0
+  ) {
+    modelInputField = model.input.field_names[0];
+  }
+  const modelType = model.model_type;
+  const modelVersion = model.version;
+  return {
+    description: '',
+    version: 1,
+    processors: [
+      {
+        remove: {
+          field: `ml.inference.${destinationField}`,
+          ignore_missing: true,
+        },
+      },
+      {
+        inference: {
+          model_id: `${modelId}`,
```

Comment:
Do you need the interpolation?
```suggestion
          model_id: modelId,
```